src/Text/Pandoc/Writers/Markdown/Inline.hs

```diff
@@ -396,60 +450,77 @@ inlineToMarkdown opts (Quoted DoubleQuote lst) = do
-             (if variant == PlainText
-                 then id
-                 else escapeText opts) $ str
+  let str' = case variant of
+                Markua -> escapeMarkuaString str
+                _ -> (if writerPreferAscii opts
+                        then toHtml5Entities
+                        else id) .
+                     (if isEnabled Ext_smart opts
+                        then unsmartify opts
+                        else id) .
+                     (if variant == PlainText
+                        then id
+                        else escapeText opts) $ str
   return $ literal str'
-inlineToMarkdown opts (Math InlineMath str) =
-  case writerHTMLMathMethod opts of
-       WebTeX url -> inlineToMarkdown opts
-                       (Image nullAttr [Str str] (url <> urlEncode str, str))
-       _ | isEnabled Ext_tex_math_dollars opts ->
-             return $ "$" <> literal str <> "$"
-         | isEnabled Ext_tex_math_single_backslash opts ->
-             return $ "\\(" <> literal str <> "\\)"
-         | isEnabled Ext_tex_math_double_backslash opts ->
-             return $ "\\\\(" <> literal str <> "\\\\)"
-         | otherwise -> do
-             variant <- asks envVariant
-             texMathToInlines InlineMath str >>=
-               inlineListToMarkdown opts .
-                 (if variant == PlainText then makeMathPlainer else id)
-inlineToMarkdown opts (Math DisplayMath str) =
-  case writerHTMLMathMethod opts of
-      WebTeX url -> (\x -> blankline <> x <> blankline) `fmap`
-             inlineToMarkdown opts (Image nullAttr [Str str]
-                    (url <> urlEncode str, str))
-      _ | isEnabled Ext_tex_math_dollars opts ->
-            return $ "$$" <> literal str <> "$$"
-        | isEnabled Ext_tex_math_single_backslash opts ->
-            return $ "\\[" <> literal str <> "\\]"
-        | isEnabled Ext_tex_math_double_backslash opts ->
-            return $ "\\\\[" <> literal str <> "\\\\]"
-        | otherwise -> (\x -> cr <> x <> cr) `fmap`
-            (texMathToInlines DisplayMath str >>= inlineListToMarkdown opts)
+inlineToMarkdown opts (Math InlineMath str) = do
+  variant <- asks envVariant
+  if variant == Markua
+      then return $ "`" <> literal str <> "`" <> "$"
+      else
+        case writerHTMLMathMethod opts of
+          WebTeX url -> inlineToMarkdown opts
+                          (Image nullAttr [Str str] (url <> urlEncode str, str))
+          _ | isEnabled Ext_tex_math_dollars opts ->
+                return $ "$" <> literal str <> "$"
+            | isEnabled Ext_tex_math_single_backslash opts ->
+                return $ "\\(" <> literal str <> "\\)"
+            | isEnabled Ext_tex_math_double_backslash opts ->
+                return $ "\\\\(" <> literal str <> "\\\\)"
+            | otherwise -> 
+                texMathToInlines InlineMath str >>=
+                  inlineListToMarkdown opts .
+                    (if variant == PlainText then makeMathPlainer else id)
```

Comment:
This is OK, but it might be nicer to avoid this nesting and just have a case statement, with the first case being:
```hs
  _ | variant == Markua -> return $ "`" etc.
```

Reply:
we changed it

Reply:
Thank You, we appreciate your fast feedback and we adjusted the code accordingly.